### PR TITLE
terminal: improve option help message

### PR DESCRIPTION
**Pull request by varmanishant**
#### What it does
Improves option help message formatting for #7284 .

#### How to test
 Open the user JSON and hover over the respective settings.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- [x] as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
@varmanishant thank you, before we can accept the change please be sure to [sign-off](https://github.com/eclipse-theia/theia/blob/master/CONTRIBUTING.md#sign-your-work) and also complete the [Eclipse Contributor Agreement (ECA)](https://github.com/eclipse-theia/theia/blob/master/CONTRIBUTING.md#eclipse-contributor-agreement).

File: packages/terminal/src/browser/terminal-preferences.ts [hunk 1]
```diff
@@ -97,12 +97,12 @@ export const TerminalConfigSchema: PreferenceSchema = {
...
         },
         'terminal.integrated.cursorStyle': {
-            description: 'Controls the style of the terminal cursor/',
+            description: 'Controls the style of the terminal cursor.',
             enum: ['block', 'underline', 'line'],
             default: 'block'
         },
         'terminal.integrated.cursorWidth': {
-            description: 'Controls the width of the cursor when \'cursorStyle\' is set to \'line\'.',
+            markdownDescription: 'Controls the width of the cursor when `#terminal.integrated.cursorStyle` is set to `line`.',
```
Comment (marechal-p) [inline packages/terminal/src/browser/terminal-preferences.ts]: What's the `#` for?

**Comment by varmanishant** [comment]
Done @vince-fugnitto . Please let me know if there is anything missing.

File: packages/terminal/src/browser/terminal-preferences.ts (hunk 1 above)
Comment (varmanishant) [reply packages/terminal/src/browser/terminal-preferences.ts]: Thanks for the question. It is a link AFAIK. I took reference from Visual Studio Code. Do you want me to link that?

Comment (marechal-p) [reply packages/terminal/src/browser/terminal-preferences.ts]: You should not remove the `description` field and strip markdown element for it. We use `description` when hovering over the tree view. You patch improves things when in the editor, but not when using the tree-based UI.

Comment (varmanishant) [reply packages/terminal/src/browser/terminal-preferences.ts]: Thanks @marechal-p , added it back. Let me know if there is anything missing.

File: packages/terminal/src/browser/terminal-preferences.ts [hunk 2]
```diff
@@ -97,12 +97,13 @@ export const TerminalConfigSchema: PreferenceSchema = {
             default: false
         },
         'terminal.integrated.cursorStyle': {
-            description: 'Controls the style of the terminal cursor/',
+            description: 'Controls the style of the terminal cursor.',
             enum: ['block', 'underline', 'line'],
             default: 'block'
         },
         'terminal.integrated.cursorWidth': {
             description: 'Controls the width of the cursor when \'cursorStyle\' is set to \'line\'.',
```
Comment (marechal-p) [inline packages/terminal/src/browser/terminal-preferences.ts]: nitpick, but while we are at it let's also fully qualify `cursorStyle` in the `description` field:
```ts
description: 'Controls the width of the cursor when \'terminal.integrated.cursorStyle\' is set to \'line\'.',
```

Comment (marechal-p) [reply packages/terminal/src/browser/terminal-preferences.ts]: Should be good to merge once this is done, I know nitpicks are annoying to address :)

Comment (varmanishant) [reply packages/terminal/src/browser/terminal-preferences.ts]: I have done the changes. Testing is annoying :-).

**Pull request merged**